numpy/core/src/common/lowlevel_strided_loops.h

```diff
@@ -752,15 +733,16 @@ PyArray_EQUIVALENTLY_ITERABLE_OVERLAP_OK(PyArrayObject *arr1, PyArrayObject *arr
     return (!arr1_read || arr1_ahead) && (!arr2_read || arr2_ahead);
 }
 
+/* This macro requires one of arr1 and arr2 to be 1-D or contiguous. */
 #define PyArray_EQUIVALENTLY_ITERABLE_BASE(arr1, arr2) (            \
                         PyArray_NDIM(arr1) == PyArray_NDIM(arr2) && \
                         PyArray_CompareLists(PyArray_DIMS(arr1), \
                                              PyArray_DIMS(arr2), \
                                              PyArray_NDIM(arr1)) && \
-                        (PyArray_FLAGS(arr1)&(NPY_ARRAY_C_CONTIGUOUS| \
-                                      NPY_ARRAY_F_CONTIGUOUS)) & \
-                                (PyArray_FLAGS(arr2)&(NPY_ARRAY_C_CONTIGUOUS| \
-                                              NPY_ARRAY_F_CONTIGUOUS)) \
+                        ((PyArray_FLAGS(arr1) & (NPY_ARRAY_C_CONTIGUOUS| \
+                                                 NPY_ARRAY_F_CONTIGUOUS)) == \
```

Comment:
Hmm, I will undo this... Its a bit tricky (correct, but its not really better as is, and the points in the code still using it should maybe just be modified).

The point is that if the arrays are 1-D, they don't have to be contiguous at all, so this check could be skipped entirely if both are 1-D.  My new code does this, which makes things like:
```
arr_1d[::2] + arr1_d[1::2]
```
or any other such operation where one of the operands is 1-D strided *much* faster (mainly lower overhead, but that can be a factor of 2 for small additions).  Making it `==` works for the above, but won't work for:
```
arr_1d[::2] + other_arr_1d_contiguous
```
since then the contiguity doesn't match... that is why I think I will undo it to avoid confusion.